client-skeleton-publisher/src/main/java/eu/arrowhead/client/skeleton/publisher/PublisherMain.java

```diff
@@ -0,0 +1,100 @@
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+
+import eu.arrowhead.client.library.ArrowheadService;
+import eu.arrowhead.client.library.util.ClientCommonConstants;
+import eu.arrowhead.client.skeleton.publisher.event.PresetEventType;
+import eu.arrowhead.common.CommonConstants;
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.dto.shared.EventPublishRequestDTO;
+import eu.arrowhead.common.dto.shared.SystemRequestDTO;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.common.exception.AuthException;
+import eu.arrowhead.common.exception.BadPayloadException;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.exception.UnavailableServerException;
+
+@SpringBootApplication
+@ComponentScan(basePackages = {CommonConstants.BASE_PACKAGE}) //TODO: add custom packages if any
+public class PublisherMain implements ApplicationRunner {
+
+    //=================================================================================================
+	// members
+	
+	@Value(ClientCommonConstants.$CLIENT_SYSTEM_NAME)
+	private String clientSystemName;
+	
+	@Value(ClientCommonConstants.$CLIENT_SERVER_ADDRESS_WD)
+	private String clientSystemAddress;
+	
+	@Value(ClientCommonConstants.$CLIENT_SERVER_PORT_WD)
+	private int clientSystemPort;
+	
+	@Autowired
+	ArrowheadService arrowheadService;
+	
+	private final Logger logger = LogManager.getLogger(PublisherApplicationInitListener.class);
+	
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public static void main(final String[] args) {
+		SpringApplication.run(PublisherMain.class, args);
+	}
+
+	@Override
+	public void run(ApplicationArguments args) throws Exception {
+		logger.debug( "run started..." );
+		
+		publishRunStartedEvent();
+		
+	}
+	
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	//Sample implementation of event publishing when application run started
+	private void publishRunStartedEvent() {
+		logger.debug( "publishRunStartedEvent started..." );
+		
+		final String eventType = PresetEventType.START_RUN.getEventTypeName();
+		
+		final SystemRequestDTO source = new SystemRequestDTO();
+		source.setSystemName( clientSystemName );
+		source.setAddress( clientSystemAddress );
+		source.setPort( clientSystemPort );
+		source.setAuthenticationInfo( Base64.getEncoder().encodeToString( arrowheadService.getMyPublicKey().getEncoded() ) );
```

Comment:
getMyPublicKey() can be null, if ssl is not enabled. Set authenticationInfo only if ssl is enabled

Reply:
fixed